agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryChannel.java

```diff
@@ -364,19 +364,21 @@ private void handleStatsbeatOnResponse(
       String instrumentationKey, Consumer<Boolean> onFailure, OperationLogger operationLogger) {
 
     return error -> {
-      if (isStatsbeat
-          && (error instanceof UnknownHostException
-              || error instanceof UnresolvedAddressException)) {
-        // when sending a Statsbeat request and server returns an UnknownHostException, it's
-        // likely that it's using AMPLS. In that case, we use the kill-switch to turn off Statsbeat.
+      if (isStatsbeat) {
```

Comment:
If you shutdown Statsbeat with any error you will see a huge drop of the data, Breeze returning errors is quite normal, we made similar changes in Node.js SDK [here](https://github.com/microsoft/ApplicationInsights-node.js/pull/922) and I actually updated the logic as I saw a significant drop of Statbeat with the specific version with that change [here](https://github.com/microsoft/ApplicationInsights-node.js/pull/924)

Reply:
we only shut down statsbeat when sending statsbeat request. it doen't impact customers' telemetry though. we have seen that AMPL will run into different exceptions for different apps. we need to shut it down until we can have a definite way of detecting AMPL or any other private endpoints.

Reply:
@heyams what if we only shut down statsbeat if the first request fails?

(and still suppress logging always for statsbeat errors)

Reply:
What I'm saying is that you will get a huge drop of Statsbeat data for all customers even in not AMPL because network error is quite common for Breeze, not talking about customer telemetry here, @trask suggestion could work as is more about the probability of first Breeze call to fail, I will argue you can also expect a drop of Statsbeat in that case but definitely better, maybe a 5 in a row fails and then disable.

Reply:
shut down after 3 errors in a row.